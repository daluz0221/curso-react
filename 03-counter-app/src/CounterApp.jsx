import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState(value)

    const handlAdd = () => {
        // console.log(event);
        // setCounter( counter + 1 );
        setCounter( (e)=> e + 1)
    }
    const handlSubstrac = () => {    
        if (counter == 0) return;
        setCounter( (e)=> e - 1)
    }

    const handlReset = () => {
        setCounter( value)
    }

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button
            onClick={  handlSubstrac }
            aria-label='btn-resta'
        >
            -1
        </button>

        <button
            onClick={  handlReset }
            aria-label='btn-reset'
        >
            Reset
        </button>

        <button
            onClick={  handlAdd }
            aria-label='btn-suma'
        >
            +1
        </button>
    </>
  )
}


CounterApp.propTypes= {
    value: PropTypes.number.isRequired
}