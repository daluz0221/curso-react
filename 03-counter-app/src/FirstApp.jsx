import PropTypes from 'prop-types'


export const FirstApp = ( {title, subtitle} ) => {


    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
        </>
  )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    // title: 'No hay texto que mostra en title',
    subtitle: 'No hay texto que mostra en subtitle'
}
