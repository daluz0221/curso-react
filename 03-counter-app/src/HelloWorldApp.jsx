

export function App() {
    return  <h1>Hol mundo</h1>
}