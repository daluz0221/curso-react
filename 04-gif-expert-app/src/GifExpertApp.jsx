import { useState } from "react"
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Naruto']);

    const onAddCategory = (event) => {
        if (categories.includes(event)) {
            return;
        }

      setCategories( [event, ...categories] )
    }

  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>

        {/* Input */}
        <AddCategory 
            // onAddCategory={setCategories} 
            onNewCategory={ onAddCategory }    
        />

        {/* Listado de gifs */}
        
        {
            categories.map( category => (
                    <GifGrid category={category} key={category}/>
                )
                )
        }
          
    </>
  )
}
