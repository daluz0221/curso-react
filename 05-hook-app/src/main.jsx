import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/MainApp'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { SimpleFormWithCustomHook } from './02-useEffect/SimpleFormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { Memorize } from './05-memos/Memorize'
// import { MemorizeHook } from './06-useMemo/MemorizeHook'
// import { CallbackHook } from './06-useMemo/CallbackHook'
// import { Padre } from './07-tarea-memo/Padre'
// import { HooksApp } from './HooksApp.jsx'

// import './08-useReducer/intro-reducer';

import { BrowserRouter } from 'react-router-dom'




ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
      <MainApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
)
