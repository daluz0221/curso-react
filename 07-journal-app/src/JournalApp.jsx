import React from 'react'
import { Approuter } from './router/Approuter'
import { AppTheme } from './theme/AppTheme'

export const JournalApp = () => {
  return (
    <AppTheme>
        <Approuter />
    </AppTheme>
  )
}
