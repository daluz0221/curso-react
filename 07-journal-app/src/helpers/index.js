

export * from './loadNotes';
export * from './getEnvironments'