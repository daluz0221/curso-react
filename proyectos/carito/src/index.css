button {
  width: 100%;
}


.span-price {
  margin-bottom: 5px;
  padding-bottom: 10px;
  font-size: 1.5rem;
}

.price-low {
  font-size: 1rem;
  color: rgb(161, 161, 161);
}

.percent-discount{
  background-color: rgb(255, 255, 88);
  border-radius: 30%;
  width: 3rem;
  display: inline-block;
  margin-left: .5rem;
  font-size: .8rem;
  text-align: center;
}

.new-price {
  font-size: 1.5rem;
}

.subCategory-filter {
  margin-left: 1rem;
}