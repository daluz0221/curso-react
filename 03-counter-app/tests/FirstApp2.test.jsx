import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp', () => {  

    const title = 'Hola, soy pipe'
    const subtitle = 'soy un subtitulo'

    test('Debe hacer match con el snapshot', () => { 

        const { container } = render( <FirstApp title={ title } /> )
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar el mensaje "Hola, soy pipe"', () => { 

        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();

    });

    test('Debe mostrar el titulo en un h1', () => { 

        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { leve: 1 }).innerHTML ).toContain( title );

    });

    test('Debe mostrar el subtitulo enviado por props', () => { 


        render( <FirstApp title={ title } subtitle={subtitle} /> )
        expect( screen.getAllByText(subtitle).length ).toBe(1);
    });

});