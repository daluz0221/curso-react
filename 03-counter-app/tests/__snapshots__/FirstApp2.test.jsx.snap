// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy pipe
  </h1>
  <p>
    No hay texto que mostra en subtitle
  </p>
</div>
`;
